src/client-functions/selector-builder/add-api.js

```diff
@@ -0,0 +1,313 @@
+import { ELEMENT_SNAPSHOT_PROPERTIES, NODE_SNAPSHOT_PROPERTIES } from './snapshot-properties';
+import { CantObtainInfoForElementSpecifiedBySelectorError } from '../../errors/test-run';
+import getCallsite from '../../errors/get-callsite';
+import ClientFunctionBuilder from '../client-function-builder';
+import {
+    assertStringOrRegExp,
+    assertNonNegativeNumber,
+    assertFunctionOrString,
+    assertFunctionOrStringOnNonNegativeNumber
+} from '../../errors/runtime/type-assertions';
+
+const SNAPSHOT_PROPERTIES = NODE_SNAPSHOT_PROPERTIES.concat(ELEMENT_SNAPSHOT_PROPERTIES);
+
+
+var filterNodes = (new ClientFunctionBuilder((nodes, filter, querySelectorRoot) => {
+    if (typeof filter === 'number')
+        return nodes[filter];
+
+    var result = [];
+
+    if (typeof filter === 'string') {
+        // NOTE: we can search for elements only in document or element.
+        if (querySelectorRoot.nodeType !== 1 && querySelectorRoot.nodeType !== 9)
+            return null;
+
+        var matching    = querySelectorRoot.querySelectorAll(filter);
+        var matchingArr = [];
+
+        for (var i = 0; i < matching.length; i++)
```

Comment:
We can use `toArray` method https://github.com/DevExpress/testcafe/blob/master/src/client/core/utils/array.js#L10